### PR TITLE
2023.12.1 documentation changes

### PR DESCRIPTION
**Pull request by ntoll**
**DO NOT MERGE UNTIL 2023.12.1 IS RELEASED**

* Fixes broken links.
* Updates docs to reflect changes from older versions.
* Bump version to 2023.12.1
* Revise terminal documentation.
* Add Python editor documentation.

### CODE REVIEW

File: docs/user-guide/editor.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+Finally, it is possible to specify a target node into which the code editor is
+created:
+
+```html title="Specify a target for the Python editor."
+<script type="mpy-editor" target="editor">
+  import sys
+  print(sys.version)
+</script>
+<div id="editor"></div> <!-- will eventually contain the Python editor -->
```
Comment (WebReflection) [inline docs/user-guide/editor.md:72]: will eventually contain the output, not the editor ... the editor is `@codemirror` matter, where the output land, the target for `print` statements, is what the `target` attribute refers to.

**Comment by WebReflection** [comment]
A minor typo or misleading comment a part, the rest LGTM :+1:

File: docs/user-guide/editor.md (hunk 1 above)
Comment (ntoll) [reply docs/user-guide/editor.md:72]: Good catch. Will fix now.

Comment (WebReflection) [reply docs/user-guide/editor.md:72]: My deep apologies I've just checked the code and you were right ... the `target` as it was with classic is *where* the editor should show, I feel like we miss an `output` attribute at this point for where the editor output lands ... this is all new to me, I've messed up my previous comment as I was confused myself.

Comment (ntoll) [reply docs/user-guide/editor.md:72]: NP - I'll revert the change, but I like the `output` attribute... this what refinement looks like. ;-)

**Pull request merged**